src/GogeDao.sol

```diff
@@ -370,718 +476,955 @@ contract GogeDAO is Ownable {
+        require(gateKeeper[_addr] != _value, "Already set");
+        gateKeeper[_addr] = _value;
     }
 
-    // ---------- Admin ----------
+    /// @notice An internal method for updating quorum value.
+    /// @param  _amount quorum value.
+    function _updateQuorum(uint256 _amount) internal {
+        require(_amount <= 100 && _amount > 0, "_amount must be between 0 and 100");
+        quorum = _amount;
+    }
 
-    function _transferOwnership(address newOwner) public onlyOwner() {
-        require(newOwner != address(0));
-        super.transferOwnership(newOwner);
+    /// @notice An internal method for updating minPollPeriod value.
+    /// @param  _amount minPollPeriod value.
+    function _updateMinPollPeriod(uint256 _amount) internal {
+        minPeriod = _amount;
     }
 
-    function payTeam() public {
-        uint256 amount = teamBalance.div(teamMembers.length);
-        for(uint256 i = 0; i < teamMembers.length.sub(1); i++) {
-            payable(teamMembers[i]).transfer(amount);
-            teamBalance -= amount;
-        }
-        payable(teamMembers[teamMembers.length.sub(1)]).transfer(teamBalance);
+    /// @notice An internal method for updating governanceTokenAddr.
+    /// @param  _addr new governance token address.
+    function _changeGovernanceToken(address _addr) internal {
+        governanceTokenAddr = _addr;
     }
 
-    // ---------- Mutative -----------
+    // ----
+    // View
+    // ----
 
-    function _updateVetoEnabled(bool enabled) internal {
-        require(veto != enabled, "Already set");
-        veto = enabled;
-        emit VetoEnabledUpdated(enabled);
+    /// @notice A view method for returning the amount of votes of a voter in a poll.
+    /// @param  _addr address of voter.
+    /// @param _pollNum unique poll identifier.
+    function getVotes(address _addr, uint256 _pollNum) public view returns (uint256) {
+        return polls[_pollNum][_addr];
     }
 
-    function updateVetoEnabled(bool enabled) external onlyOwner() {
-        _updateVetoEnabled(enabled);
+    /// @notice A view method for returning a poll's unique metadata.
+    /// @param _pollNum unique poll identifier.
+    function getMetadata(uint256 _pollNum) public view returns (Metadata memory) {
+        return pollMap[_pollNum];
     }
 
-    function setTeamMember(address addr, bool value) internal {
-        if(value) {
-            (bool _isTeamMember, ) = isTeamMember(addr);
-            if(!_isTeamMember) teamMembers.push(addr);        
-        } else {
-            (bool _isTeamMember, uint8 s) = isTeamMember(addr);
-            if(_isTeamMember){
-                teamMembers[s] = teamMembers[teamMembers.length - 1];
-                teamMembers.pop();
-            } 
+    /// @notice A view method for returning a poll's current proportion of votes over circuating supply.
+    /// @param _pollNum unique poll identifier.
+    function getProportion(uint256 _pollNum) public view returns (uint256) {
```

Comment:
Why is an extra public method necessary to determine the proportion?

Reply:
I found myself calculating the proportion tooo many times in my tests so just having a function I could call for a specific poll helped a ton. I could've made this a utility in my tests but I found this is also very helpful in getting the proportion for the front end when it displays the positive votes for a poll